torchrl/modules/models/decision_transformer.py

```diff
@@ -0,0 +1,295 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+from __future__ import annotations
+
+import dataclasses
+
+import importlib
+from dataclasses import dataclass
+from typing import Any, Optional
+
+import torch
+import torch.nn as nn
+
+_has_transformers = importlib.util.find_spec("transformers") is not None
+import transformers
+from transformers.models.gpt2.modeling_gpt2 import (
+    BaseModelOutputWithPastAndCrossAttentions,
+    GPT2Model,
+)
+
+
+class ModifiedGPT2Model(GPT2Model):
```

Comment:
Wrapper class to remove wpe layer of the GPT2Model from transformers. Maybe we can compress this even more?

Reply:
1. This should run even if transformers isn't installed. 
2. Do we have dedicated tests?
3. Is it integrated in the doc?
4. The docstring is a bit cryptic for someone who doesn't know what it is all about.
I wish transformers had a more modular code... What is the signature of wpe? In some cases we can simply replace the layer by nn.Identity()...

Reply:
I tried to use the identity but got some shape issues. But I found out that with all the fixes I did it now even converges with the wpe layer. For comparison, I also ran a test where I exchanged the wpe layer with a custom ZeroPosEmbeddingLayer returning only zeros. In the graph, you can see with wpe and with zero wpe. 
Let me know what you think. For now, I took the ZeroPosEmbeddingLayer off as it does converge but I can add it as well.

![image](https://github.com/pytorch/rl/assets/29492081/936d9cc1-c5bc-422e-9bf7-5a941a4c56a4)

Reply:
Oh but at this point let's get rid of that class altogether no?

Reply:
Yes, I removed it all already. If you can have a final look I think it should be ready now.